src/coreclr/vm/comutilnative.cpp

```diff
@@ -1827,6 +1827,13 @@ NOINLINE static FC_BOOL_RET CanCompareBitsHelper(MethodTable* mt, OBJECTREF objR
     FC_RETURN_BOOL(ret);
 }
 
+FCIMPL1(UINT32, ValueTypeHelper::GetObjectSize, Object* obj)
+{
+    FCALL_CONTRACT;
+    return obj->GetTypeHandle().GetSize();
+}
+FCIMPLEND
```

Comment:
Questions:
- Signature: UINT32 is what `GetSize` returns. Should we change to `nuint` or other?
- Parameter: I haven't seen any `MethodTable*` or `TypeHandle` in FCall. Passing an object looks like the common way.
- Implementation: The `TypeDesc` path is not used. Could it be changed to call `MethodTable::GetNumInstanceFieldBytes()` directly? What about the correctness of `CorElementType` path?
- The `ParentMethodTable` member definition looks like to match the union with `EEClass` ptr. If we add `EEClass` definition in managed, we can access it in managed code then.

Reply:
> Signature: UINT32 is what GetSize returns. Should we change to nuint or other?
Parameter: I haven't seen any MethodTable* or TypeHandle in FCall. Passing an object looks like the common way.

UINT32 as it is sounds good to me.

> Implementation: The TypeDesc path is not used. Could it be changed to call MethodTable::GetNumInstanceFieldBytes() directly?

You can consider changing the FCall to be take `MethodTable*` instead of object and expose it as an exact mirror the unmanaged MethodTable. The managed can then look like this:

`return SpanHelpers.SequenceEqual(..., RuntimeHelpers.GetMethodTable(obj)->GetNumInstanceFieldBytes());`

> What about the correctness of CorElementType path?

I am not sure what you mean.

> How many cases would benefit if we add EEClass definition in managed?

EEClass is definition is pretty complicated. EEClass contains less performance critical data. I think it is fine to fetch data from EEClass using FCalls rather than maintain the duplicated definition.

Reply:
As my knowledge, calling convention for struct instance method is `arg0 := &this`, so the new definition is correct? I'm still quite hesitate since no existing FCall does this.

Reply:
It looks good to me.